Automattic/wp-calypso
# Allow copying sites on Commerce plans: Fix hardcoded Business plan on Calypso

Related to:
- https://github.com/Automattic/wp-calypso/issues/96383

## Proposed Changes

* We now supply the plan of the site we're copying.
* We have a plan whitelist, so that you can only checkout a Business or Commerce plan when copying.

## Why are these changes being made?


* On Calypso the plan we use when copying a site is fixed to the Business plan, this isn't always correct since we can copy an E-Commerce site and we should checkout a commerce site instead of a business site.

## Testing Instructions



* Open the Calypso live link.
* Copy an E-Commerce site.
* The checkout page should include the Commerce plan instead of the Business plan
* Try a site on a Business plan, there should be no regressions.

https://github.com/user-attachments/assets/55004a09-2499-4c52-906b-fa865edbb7cd

## Pre-merge Checklist
